src/common/file_system/virtual_file_system.cpp

```diff
@@ -0,0 +1,85 @@
+#include "common/file_system/virtual_file_system.h"
+
+#include "common/file_system/local_file_system.h"
+
+namespace kuzu {
+namespace common {
+
+VirtualFileSystem::VirtualFileSystem() {
+    defaultFS = std::make_unique<LocalFileSystem>();
+}
+
+FileSystem* VirtualFileSystem::findFileSystem(const std::string& /*path*/) {
+    return defaultFS.get();
+}
+
+std::unique_ptr<FileInfo> VirtualFileSystem::openFile(
+    const std::string& path, int flags, FileLockType lockType) {
+    return findFileSystem(path)->openFile(path, flags, lockType);
+}
+
+void VirtualFileSystem::readFromFile(
+    FileInfo* fileInfo, void* buffer, uint64_t numBytes, uint64_t position) {
+    fileInfo->fileSystem->readFromFile(fileInfo, buffer, numBytes, position);
+}
+
+int64_t VirtualFileSystem::readFile(FileInfo* fileInfo, void* buf, size_t nbyte) {
+    return fileInfo->fileSystem->readFile(fileInfo, buf, nbyte);
+}
+
+void VirtualFileSystem::writeFile(
+    FileInfo* fileInfo, const uint8_t* buffer, uint64_t numBytes, uint64_t offset) {
+    fileInfo->fileSystem->writeFile(fileInfo, buffer, numBytes, offset);
+}
+
+int64_t VirtualFileSystem::seek(FileInfo* fileInfo, uint64_t offset, int whence) {
+    return fileInfo->fileSystem->seek(fileInfo, offset, whence);
+}
+
+std::vector<std::string> VirtualFileSystem::glob(const std::string& path) {
+    return findFileSystem(path)->glob(path);
+}
+
+void VirtualFileSystem::overwriteFile(const std::string& from, const std::string& to) {
+    findFileSystem(from)->overwriteFile(from, to);
+}
+
+void VirtualFileSystem::copyFile(
```

Comment:
Double check here, too, codecov complains this function is also not used. Same for `removeDir`, `truncate`, and `getFileSize`.

Reply:
removed this function